### DISCUSSION TITLE
redirectTo not respected when resetting password with https://

### DISCUSSION THREAD

**Discussion by waqas1739** [supabase/supabase#21972] [Questions]
When i try the password reset with the localhost, after getting the email confirmation, i get redirected to /xys which is the expected behaviour

When i add redirect to https/. it shows the Site_URL with # and no redirect route or shows the access token and 404.  Trying through documentation but did not seem to work.

Any solution?

**Comment by GaryAustin1** [comment]
Are you using redirectTo or emailRedirectTo in the API call?

**Reply by waqas1739** [reply]
Hey Gary I can add here my method for reset and update and you can give me a hint if there is a mistake

**Reply by GaryAustin1** [reply]
It would help if you showed your call, yes.

**Comment by waqas1739** [comment]
Check this @GaryAustin1   
Password reset: 
const handleSendRequest = async () => {
  setIsLoading(true);
  try {
    if (supabase) {
      
      // Construct the new URL without the access token and other parameters
      const redirectUrl = `${window.location.origin}/newpassword`;
      console.log("Reset Password Link:", redirectUrl);
      const { error } = await supabase.auth.resetPasswordForEmail(
        email,
        { redirectTo: redirectUrl }
      );
      if (error) {
        throw error;
      }
      setOpenDialog(true);
    }
  } catch (error: any) {
    console.error("Error sending password reset request:", error.message);
    navigate("/resetpassword");
  } finally {
    setIsLoading(false);
  }
};


Update:
  const handleUpdatePassword = async () => {
    try {
        setIsLoading(true);
        if (supabase) {
            const { data: currentUser, error } = await supabase.auth.getUser();
            if (error) {
                throw error;
            }
            
            // Proceed only if a user exists
            if (currentUser) {
                // Update the password using Supabase
                await supabase.auth.updateUser({ password });
                setOpenDialog(true);
            } else {
                throw new Error("User does not exist.");
            }
        }
    }  catch (error: any) {
        console.error("Password update error:", error.message);
    }
    finally {
        setIsLoading(false);
    }
};

**Reply by GaryAustin1** [reply]
`${window.location.origin}/newpassword`

Is that exact location (full URL) in the redirect URL settings in the dashboard?

If it redirects to your siteURL setting then the redirectTo URL likely is not matched in the dashboard settings exactly.  Otherwise something with your framework is impacting the redirect (which is not unheard of, but I'm no much help with that).

**Reply by waqas1739** [reply]
No. In the URL configuration in supabase in Redirect URL it should hit eg "https://abc.com/** " to allow redirect /newpassword

Am i doing mistake here or in the email template and url configuration?

**Reply by AkshayAdiga1997** [reply]
@waqas1739 
Go to your site, and in the console, try `window.origin`. 
Use the same in the URL list. For example, it is says https://abc.com, use the same in the URL list and not https://www.abc.com (exact match is needed). 
So the final URL in the list will be https://abc.com/xyz and not https://www.abc.com/xyz
